src/diff/index.js

```diff
@@ -232,6 +232,15 @@ export function diff(
 
 		if ((tmp = options.diffed)) tmp(newVNode);
 	} catch (e) {
+		// When we are hydrating and have excessDomChildren we don't know the _children this VNode
+		// should receive so it's safer to unmount them. Else on the subsequent error-boundary diff,
+		// we won't know the oldDom and insert an additional node instead of replace the prerendered one. (#2539)
+		if (isHydrating && excessDomChildren != null) {
+			for (tmp = excessDomChildren.length; tmp--; ) {
```

Comment:
We could pass `excessDomChildren` to `options._catchError` and only do this if an error-boundary catches the error. Not entirely sure if that's better.